File: app/Subscribe.c
#include <stdio.h>
#include "MMW.h"

void test_callback(const char *topic, const char *message) {
    printf("Got message in callback: %s\n", message);
}

int main() {

    // Initialize library settings
    if (mmw_initialize("127.0.0.1", 5001, MMW_TRANSPORT_WEBSOCKET) != MMW_OK) {
        return -1;
    }

    // Test API call
    if (mmw_create_subscriber("Test Topic", test_callback) != MMW_OK) {
        return -1;
    }

    // Sleep to keep subscriber up
    mmw_wait();

    // Cleanup
    if (mmw_cleanup() != MMW_OK) {
        return -1;
    }
}

File: python/app/subscribe.py
#!/usr/bin/python3
import mmw
import time
import signal
import sys

def my_callback(topic: str, msg: str):
    print(f"[Python Callback] Received: {msg}")

def my_callback_2(topic: str, msg: str):
    print(f"[Python Callback] Received: {msg}")

# Set the log level
# mmw.set_log_level(mmw.MmwLogLevel.MMW_LOG_LEVEL_OFF)

# Initialize the library
mmw.initialize("127.0.0.1", 5000)

# Create a subscriber object
subscriber = mmw.create_subscriber("Test Topic", my_callback)
subscriber_2 = mmw.create_subscriber("Test Topic 2", my_callback_2)

print("Waiting for messages... Ctrl+C to exit")

# Keep the main thread alive
while True:
    time.sleep(1)
